keras_cv/src/models/object_detection/mask_rcnn/mask_rcnn.py

```diff
@@ -0,0 +1,502 @@
+# Copyright 2024 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import tensorflow as tf
+import tree
+
+from keras_cv.src.api_export import keras_cv_export
+from keras_cv.src.backend import keras
+from keras_cv.src.backend import ops
+from keras_cv.src.bounding_box import convert_format
+from keras_cv.src.bounding_box.converters import _decode_deltas_to_boxes
+from keras_cv.src.bounding_box.utils import _clip_boxes
+from keras_cv.src.layers.object_detection.roi_sampler import ROISampler
+from keras_cv.src.models.object_detection.faster_rcnn import MaskHead
+from keras_cv.src.models.object_detection.faster_rcnn.faster_rcnn import (
+    BOX_VARIANCE,
+)
+from keras_cv.src.models.object_detection.faster_rcnn.faster_rcnn import (
+    _parse_box_loss,
+)
+from keras_cv.src.models.object_detection.faster_rcnn.faster_rcnn import (
+    _parse_classification_loss,
+)
+from keras_cv.src.models.object_detection.faster_rcnn.faster_rcnn import (
+    _parse_rpn_classification_loss,
+)
+from keras_cv.src.models.object_detection.faster_rcnn.faster_rcnn import (
+    unpack_input,
+)
+from keras_cv.src.models.task import Task
+
+
+@keras_cv_export(
+    [
+        "keras_cv.models.MaskRCNN",
+        "keras_cv.models.object_detection.MaskRCNN",
+    ]
+)
+class MaskRCNN(Task):
+    """A Keras model implementing the Mask R-CNN architecture.
+
+    Mask R-CNN is a straightforward extension of Faster R-CNN, providing an
+    additional mask head that predicts segmentation masks. The constructor
+    therefore requires a FasterRCNN model as backend.
+
+    This model is compatible with Keras 3 only.
+
+    Args:
+         backbone: `keras.Model`. A FasterRCNN model that is used for
```

Comment:
backbone: `keras.Model`. A FasterRCNN model that is used for
            object detection. ==> backbone: `keras.Model`. A FasterRCNN backbone model.